addons/flannel/0.21.5/install.sh

```diff
@@ -239,25 +239,74 @@ function weave_to_flannel() {
     flannel_ready_spinner
 
     logStep "Restarting CSI pods"
-    kubectl -n longhorn-system delete pods --all || true
-    kubectl -n rook-ceph delete pods --all || true
-    kubectl -n openebs delete pods --all || true
+    log "this may take several minutes"
+    # We will try delete the Pods without force it
+    # However, in some cases such as where it has many addons it seems
+    # getting stuck and we force the pods deletion
+    echo "this may take several minutes"
+
+    # List of namespaces
+    namespaces=("longhorn-system" "rook-ceph" "openebs")
+
+    for ns in "${namespaces[@]}"; do
+        # Delete pods with a grace period and timeout, and ignore errors
+        timeout 80 kubectl -n ${ns} delete pods --all --grace-period=60 || logWarn "Timeout reached while trying to delete pods in namespace: ${ns}"
+
+        # Use the spinner_until function to wait until all pods are deleted
+        if ! spinner_until 60 pods_remain ${ns}; then
+            # If pods remain after the timeout, force delete them
+            logWarn "Force deleting remaining pods in namespace: ${ns}"
+            kubectl -n ${ns} delete pods --all --grace-period=0 --force || true
+        fi
+    done
 
+    echo "waiting 60 seconds before continue"
     sleep 60
+
     logStep "Restarting all other pods"
     echo "this may take several minutes"
     local ns=
-    for ns in $(kubectl get ns -o name | grep -Ev '(kube-system|longhorn-system|rook-ceph|openebs|kube-flannel)' | cut -f2 -d'/'); do
-        kubectl delete pods -n "$ns" --all --grace-period=200
+
+    # Get the list of namespaces
+    namespaces=$(kubectl get ns -o name | grep -Ev '(kube-system|longhorn-system|rook-ceph|openebs|kube-flannel)' | cut -f2 -d'/')
+
+    for ns in ${namespaces}; do
+        # Delete pods with a grace period and timeout, and ignore errors
+        timeout 80 kubectl delete pods -n "${ns}" --all --grace-period=60 || logWarn "Timeout reached while trying to delete pods in namespace: ${ns}"
+
+        # Use the spinner_until function to wait until all pods are deleted
+        if ! spinner_until 60 pods_remain "${ns}"; then
+            # If pods remain after the timeout, force delete them
+            logWarn "Force deleting remaining pods in namespace: ${ns}"
+            kubectl -n "${ns}" delete pods --all --grace-period=0 --force || true
+        fi
     done
+
+    echo "waiting 60 seconds before continue"
     sleep 60
+
     log "Awaiting up to 5 minutes to check Flannel Pod(s) are Running"
     if ! spinner_until 300 check_for_running_pods "kube-flannel"; then
         logWarn "Flannel has unhealthy Pod(s)"
     fi
     logSuccess "Migrated from Weave to Flannel"
 }
 
+# Function to check if pods are still present in a namespace
+function pods_remain() {
```

Comment:
it is our practice to prefix functions in addons and files with the name of the file to avoid collisions

```suggestion
function flannel_pods_remain() {
```